Dockerfile

```diff
@@ -129,7 +129,12 @@ ENV OPENVPN_USERNAME=**None** \
     TRANSMISSION_WEB_HOME= \
     DROP_DEFAULT_ROUTE= \
     WEBPROXY_ENABLED=false \
-    WEBPROXY_PORT=8888
+    WEBPROXY_PORT=8888 \
+    MONITORDNS_ENABLED=true 
+
+#Making the checkdns script executables
+RUN chmod 755 /etc/openvpn/monitor-checkdns.sh
+RUN chmod 755 /etc/openvpn/checkdns.sh
```

Comment:
It's weird that the `chmod` would have to be here, where is that command run for all the other scripts? (like `start.sh`?)

Reply:
> I agree with @jfessard's comments. The commits should be squashed, and chmod should be done to the file before it's added. So it's not a part of the image build.
> 
> That being said. Docker has support for native health checks. I just read a bit about them, but it seems they can be a good fit for our need. https://docs.docker.com/engine/reference/builder/#healthcheck
> 
> Basically we could make a health-check script that does DNS checks to begin with, but could also be expanded. From docker's side there is no automatic handling if the check fails (we want to exit). It was in the original PR [moby/moby#22719](https://github.com/moby/moby/pull/22719), but got removed for a future solution with restart-policy.
> 
> For now we could have "HEALTH_CHECK_ENABLED" and "HEALTH_CHECK_EXIT_ON_UNHEALTHY" variables that will decide if the tests should be run, and if an error should mark the container unhealthy or exit when it fails. This could be achieved by killing the process.
> 
> Then we would have an all-in-one solution that also kills the process, health check run as root so that should be OK. And users have the possibility to run for example this image https://github.com/willfarrell/docker-autoheal to handle the un-healthy containers.
> 
> We should also make the checks run through the tunnel device. And could we do with `dig`or `ping` instead of curl?
> 
> Thoughts?

Ok, i agree with that, i will do it but give me some times, i am actually looking for a new job :)